### PR TITLE
Feat/misc/integrate app icons

### PR DESCRIPTION
**Pull request by francelu**
# Integrate app icons into UI

## Description
This PR introduces our app icons designed by @lazarinibruno in #224 into our UI. It closes issue #226.

## Changes
- User can edit an alert and save it even though product and/or urgency hasn't been changed, which wasn't possible before.
- `data class` `PeriodPalsIcon` to create lists `LIST_OF_PRODUCTS` and `LIST_OF_URGENCIES` which have the icon as an `Int` and their corresponding `textId
- Implemented necessary changes in alerts UI, shared alert components and alert tests.
- Adapted alerts tests as necessary.

#### Added
- None.

#### Modified
- `Alert.kt`: new lists for products and urgencies icons.
- `AlertLists.kt`: adapt `AlertProductAndUrgency` and calls to it.
- `EditAlert.kt`: adapt calls to `productField()` and `urgencyField()`.
- `AlertComponents.kt`: adapt `productField()` and `urgencyField()` with an `isSelected` boolean for `EditAlertScreen`.
  `EditAlertScreen` on "Save" might not change product and urgency fields and want to save the new state.
- `AlertListsScreenTest.kt`: adapt tests with new `LIST_OF_PRODUCTS` and `LIST_OF_URGNCIES`.
- `EditAlertScreenTest.kt`: adapt tests with new `LIST_OF_PRODUCTS` and `LIST_OF_URGNCIES`.

#### Removed
- None.

## Dependencies Added
- None.

## Testing
- Adapted tests as necessary.

#### `CreateAlertScreen`
| Products  | Urgencies  |
|--|--|
| image | image |

#### `AlertListsScreen`
| My Alerts  | Pals' Alerts  |
|--|--|
| image | image |
#### `EditAlertScreen`
| `EditAlert` Product  | `EditAlert` Urgency  |
|--|--|
| image  | image  |

### CODE REVIEW

File: app/src/main/java/com/android/periodpals/model/alert/Alert.kt [hunk 1]
```diff
@@ -30,7 +32,7 @@ data class Alert(
 enum class Product {
   TAMPON,
   PAD,
-  TAMPON_AND_PAD,
+  NO_PREFERENCE,
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/model/alert/Alert.kt:35]: Good catch! This homogeneises the nomenclature, it's nice.

File: app/src/main/java/com/android/periodpals/ui/components/AlertComponents.kt [hunk 2]
```diff
@@ -38,10 +44,8 @@ import com.android.periodpals.resources.ComponentColor.getMenuTextFieldColors
 import com.android.periodpals.resources.ComponentColor.getOutlinedTextFieldColors
 import com.android.periodpals.ui.theme.dimens
 
-private val PRODUCT_DROPDOWN_CHOICES = listOf("Tampons", "Pads", "No Preference")
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/components/AlertComponents.kt:41]: Good job refactoring these properly in `Alert`

**Review by charliemangano — APPROVED** [review]
## Summary
NIce job integrating the new icons to the app! The refactoring you did also seems correct and uselul!
## Important
#### Code Quality
Good readability and code is consistent with rest of the codebase, good job!
#### Functionality
App now shows our custom icons instead of the material ones, great!
## Testing
You updated the tests properly, nice job!
## Documentation
N/A
## Suggestions
To me, some of the new icons (No Preference and Urgency levels) seem a bit unparseable since they're displayed on such a small screen. Maybe we should fill in the urgency levels/decrease the weight? For the No Preference, maybe not overlay both products but lay them side by side? In any case, I don't think this is in the scope of this PR, I'll talk about it in the meetings. 
## Steps before PR approved
LGTM!

**Comment by francelu** [comment]
I take note of the comment. I also saw that the urgency icons were a bit less UX comfortable.

File: app/src/main/java/com/android/periodpals/model/alert/Alert.kt [hunk 3]
```diff
@@ -46,3 +48,18 @@ enum class Status {
   PENDING, // Someone has acknowledged the alert and is helping
   SOLVED, // The alert has been resolved, help was provided
 }
+
+data class PeriodPalsIcon(val icon: Int, val textId: String)
```
Comment (lazarinibruno) [inline app/src/main/java/com/android/periodpals/model/alert/Alert.kt:52]: Great addition to simplify the code! Its not an absolute must, but a short documentation would be great.

**Review by lazarinibruno — APPROVED** [review]
## Summary

Its finally here! Great to see the icons in the app 🥶❤️‍🔥.

## Important

### Code Quality

- Very readable and clear.
  
- I particularly like the additons of the `PreiodPalsIcon` data class, `extractProductObject()` and `extractUrgencyObject()`. They make the code readable, modular and maintainable.
  
- You also pertinently refactored the code where needed.
  

## Testing

Coverage on new code is 86.1% :)

## Documentation

Good documentation of composables and helper functions.

## Suggestions

None. Great work as usual, France!

## Steps before PR approved

LGTM.

File: app/src/main/java/com/android/periodpals/model/alert/Alert.kt (hunk 3 above)
Comment (francelu) [reply app/src/main/java/com/android/periodpals/model/alert/Alert.kt:52]: Will do !

**Pull request merged**